### PR TITLE
Resolve numpy 1.25 issues

### PR DESCRIPTION
**Pull request by bryevdv**
This is an abridged version of #972 to clear up CI issues, that updates everything except a few more complicated uses of `find_common_types`.

A `test_argsort.py` test is failing for me locally, but also fails for me locally on `branch-23.07` so I don't think it is related.

### CODE REVIEW

**Review by manopapad — APPROVED** [review]

File: cunumeric/module.py [hunk 1]
```diff
@@ -663,7 +663,7 @@ def arange(
         step = 1
 
     if dtype is None:
-        dtype = np.find_common_type([], [type(start), type(stop), type(step)])
+        dtype = np.result_type(type(start), type(stop), type(step))
```
Comment (bryevdv) [inline cunumeric/module.py:666]: @seberg is this OK or do we need to pass actual scalars?

Comment (seberg) [reply cunumeric/module.py:666]: `find_common_type` calls `np.dtype()` first while `np.result_type` tends to convert to arrays when it doens't see an exact scalar type or dtype.  I think that could fail with the `type()` for very strange input.

So it should be a bit safer to pass the values (or use `np.dtype(type(start))` to keep the identical behavior).
Passing values may also be more future-proof, but that goes into too much `arange` weirdness.

Comment (bryevdv) [reply cunumeric/module.py]: https://github.com/nv-legate/cunumeric/pull/973/commits/30c4b26498d9acccb3cf3d9d37fa2d4b99345df5

File: cunumeric/_ufunc/ufunc.py [hunk 2]
```diff
@@ -557,33 +557,15 @@ def _find_common_type(
         if len(unique_dtypes) == 1 and all_ndarray:
             return arrs[0].dtype
 
-        # FIXME: The following is a miserable attempt to implement type
```
Comment (manopapad) [inline cunumeric/_ufunc/ufunc.py:560]: @magnatelee you might want to look over this change, as you originally put in this "emulation" code. According to @seberg here https://github.com/nv-legate/cunumeric/pull/972#discussion_r1238344237, this change should be doing the right thing.

Comment (bryevdv) [reply cunumeric/_ufunc/ufunc.py:560]: :+1: I would like some extra eyes on this

Comment (magnatelee) [reply cunumeric/_ufunc/ufunc.py:560]: I'm fine with the change as long as it passes all the tests. and I'd actually go see if we can run this test as well with the change: https://github.com/nv-legate/cunumeric/blob/branch-23.07/tests/integration/test_binary_op_typing.py#L132-L136 I vaguely remember `np.result_type` wasn't doing the right thing in the past and so I ended up with that messy typing code. if that's all sorted out in 1.25 and `np.result_type` is consistent with all the corner cases of numpy's ufuncs, it's great actually and I can't see a reason now to use it.

Comment (bryevdv) [reply cunumeric/_ufunc/ufunc.py:560]: @magnatelee all tests pass locally including those previously `xfail` tests for, except for `test_argsort.py `, butthat also fails for me on the main branch. I'd like to see this pass run in CI  once the build issues are sorted, then I will plan to merge.

Comment (seberg) [reply cunumeric/_ufunc/ufunc.py:560]: This change should be backwards and I think also future proof.  The only thing is that it converts the scalars to numpy arrays (and currently has to).

Promotion can differ for a single ufunc sometimes, I added `np.ufunc.resolve_dtypes()` for that, but it opts in to (probable) future behavior so I don't think it's time to seriously consider using it.
